mammoth/distributed/communication.py

```diff
@@ -35,81 +36,48 @@ def broadcast_tensors(tensors, src=0, group=None):
             torch.distributed.broadcast(t, src, group=group)
 
 
-def only_ready_reduce_and_rescale_grads(named_parameters, group=None):
+def managed_reduce_and_rescale_grads(
```

Comment:
"managed" ?  I'm not sure about the clarity of the terminology here, could you elaborate on that?

Reply:
We should rename this, if we can come up with a more descriptive name.

Maybe `externally_managed_reduce_and_rescale_grads` would do the trick?
The idea is that `only_ready` uses the metadata provided by the `has_grad` hook to make its own decisions about how to communicate certain parameters, but the new implementation needs to be managed by something else, with the caller passing in the `has_local_gradient` and `gradient_norm`.